[BUGFIX] Use correct module routes

jonakieling opened the pull request:
Hi there,

this fixes #1271
Some hard coded backend module routes had the /module prefix which does not work in TYPO3 9.5.16 and presumably any version of 9.5. At least I could not find any changes related to backend module routing.

This fixes our TYPO3 9.5.16 instance where the previous routes resulted in an exception being thrown. See #1271 for details.

I'm curious where the /modules comes from or where it would be needed. Maybe I'm just missing something obvious.

Kind regards
Jona

georgringer commented:
@liayn wanna test that in 10.x?

liayn commented:
The search worked flawlessly for us in the latest 8.x versions. Will try it again with this change applied.

liayn commented:
This change breaks in v10:

The requested resource "/web/NewsAdministration" was not found.

jonakieling commented:
Added a version check for getting the module path and adjusted two occurrences where the hard coded path was used.

I've checked the search form in 9.5.17 and 10.4.2. Looks good so far.

liayn commented on Classes/Backend/RecordList/RecordListConstraint.php:
```diff
@@ -31,6 +32,11 @@ class RecordListConstraint
+        if ($version < 10) {
+            return $vars === '/web/NewsAdministration/';
+        }
+
```
what about using str(r)pos() instead of a version check for this? would be much simpler here.

`return strpos('/web/NewsAdministration/', $vars) !== false;`

liayn commented:
Btw: the related core change is https://review.typo3.org/c/Packages/TYPO3.CMS/+/60433

liayn commented:
A different solution would also be to set the desired path manually, as outlined in the deprecation rst file.

liayn commented on Resources/Private/Templates/Administration/Index.html:
```diff
@@ -46,7 +46,7 @@ <h1>
 		<f:form id="administrationForm" name="demand" object="{demand}" method="get" data="{autoSubmitForm:autoSubmitForm}">
 			<input type="hidden" name="formSubmitted" value="1">
-            <input type="hidden" name="route" value="/module/web/NewsAdministration">
+            <input type="hidden" name="route" value="/module/web/NewsAdministration/">
```
the slash is too much

jonakieling replied:
Yep, just tested it with 9 LTS and 10 LTS. It is needed with 9 and breaks with 10.

jonakieling replied:
Looks like the version check is unfortunately needed after all.

jonakieling replied:
I'll move my fork to a different branch to avoid the commit spam and report back when everything is tested and working with both 9 and 10.

jonakieling replied:
The slash is assumed by the older version in https://review.typo3.org/c/Packages/TYPO3.CMS/+/60433/5/typo3/sysext/core/Classes/Utility/ExtensionManagementUtility.php#b816. I adjusted the manual module path to include the slash. After the correct TCA was loaded without cache it now works with 9 LTS and 10 LTS.

liayn commented:
Looks good by reading

jonakieling commented:
Hi @liayn 

Thanks for your support in this! After checking this with flushed TCA caches the most recent commits work with both TYPO3 9 LTS and 10 LTS.

jonakieling commented:
Latest update fixes the action buttons for creating and editing records on the upper button bar.
